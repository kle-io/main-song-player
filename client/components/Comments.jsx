/* eslint-disable react/prop-types */
/* eslint-disable class-methods-use-this */
import React from 'react';

class Comments extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      clickHover: false,
    };
    this.handleComment = this.handleComment.bind(this);
  }

  handleComment(event) {
    event.preventDefault();
    const { clickHover } = this.state;
    const user = document.getElementsByClassName('userPhoto')[0];
    if (clickHover) {
      user.style.borderRadius = '20px';
    } else {
      user.style.borderRadius = '0';
    }
    this.setState({
      clickHover: !clickHover,
    });
  }

  render() {
    const { comments } = this.props;
    const { clickHover } = this.state;
    return (
      <div className="comments">
        {comments.map((comment) => {
          return (
            <div key={comment.user} className="comment">
              <button className="photoButton" type="button" onClick={this.handleComment} onMouseOver={this.handleComment} onFocus={this.handleComment}>
                <img className="userPhoto" src={comment.photo} alt="user" />
              </button>
              {clickHover ? <div>{`${comment.user}  ${comment.comment}`}</div> : ''}
            </div>
          );
        })}
      </div>
    );
  }
}

export default Comments;
