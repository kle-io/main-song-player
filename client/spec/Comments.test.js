/* eslint-disable no-undef */
import React from 'react';
import { mount, shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import Comments from '../components/Comments';

describe('Comments', () => {
  const song = {
    artist: 'Hershel.Legros',
    genre: 'R&B & Soul',
    title: 'Handmade Frozen Soap monitor Oval',
    photo: 'https://kleiomainplayer.s3.us-west-1.amazonaws.com/covers/4.jpg',
    color1: '#243d70',
    color2: '#117572',
    duration: 219.24,
    link: 'https://kleiomainplayer.s3.us-west-1.amazonaws.com/songs/4.mp3',
    posted: '2019-11-18T08:10:56.187Z',
    comments: [
      {
        user: 'Anibal99',
        photo: 'https://kleiomainplayer.s3.us-west-1.amazonaws.com/covers/70.jpg',
        comment: 'Ut reiciendis voluptas nisi ipsum.',
        time: 130,
      },
      {
        user: 'Colten46',
        photo: 'https://kleiomainplayer.s3.us-west-1.amazonaws.com/covers/82.jpg',
        comment: 'Magnam ducimus quo optio.',
        time: 24,
      },
      {
        user: 'Linda.Simonis86',
        photo: 'https://kleiomainplayer.s3.us-west-1.amazonaws.com/covers/27.jpg',
        comment: 'Aut nemo vel soluta est laborum dolore.',
        time: 114,
      }],
  };

  it('should render all usernames, comments, photos', () => {
    const output = Renderer.create(<Comments comments={song.comments} />);
    expect(output).toMatchSnapshot();
  });
});
