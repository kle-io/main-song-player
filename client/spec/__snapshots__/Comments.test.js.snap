// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments should render all usernames, comments, photos 1`] = `
<div
  className="comments"
>
  <div
    className="comment"
  >
    <button
      className="photoButton"
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <img
        alt="user"
        className="userPhoto"
        src="https://kleiomainplayer.s3.us-west-1.amazonaws.com/covers/70.jpg"
      />
    </button>
    
  </div>
  <div
    className="comment"
  >
    <button
      className="photoButton"
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <img
        alt="user"
        className="userPhoto"
        src="https://kleiomainplayer.s3.us-west-1.amazonaws.com/covers/82.jpg"
      />
    </button>
    
  </div>
  <div
    className="comment"
  >
    <button
      className="photoButton"
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <img
        alt="user"
        className="userPhoto"
        src="https://kleiomainplayer.s3.us-west-1.amazonaws.com/covers/27.jpg"
      />
    </button>
    
  </div>
</div>
`;
